src/webots/engine/WbSimulationWorld.cpp

```diff
@@ -287,12 +295,16 @@ void WbSimulationWorld::restartStepTimer() {
 }
 
 void WbSimulationWorld::modeChanged() {
+  WbPerformanceLog *log = WbPerformanceLog::instance();
+
   const WbSimulationState::Mode mode = WbSimulationState::instance()->mode();
   switch (mode) {
     case WbSimulationState::PAUSE:
       mTimer->stop();
       WbSoundEngine::setPause(true);
       WbSoundEngine::setMute(WbPreferences::instance()->value("Sound/mute").toBool());
+      if (log)
+        log->invalidateMeasure(WbPerformanceLog::SPEED_FACTOR);
```

Comment:
Can we pause the measurement here? Or, just not stop the timer?

Reply:
You're right, I thought it would pollute the time accumulator variable if paused mid-run but I checked and it doesn't. Even pausing for a long time and resuming still yields the right result.